src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentRenderModeDirectiveIntegrationTests.cs

```diff
@@ -0,0 +1,150 @@
+        Assert.NotNull(modeProperty);
+    }
+
+    [Fact]
+    public void RenderMode_With_Static_Usings()
+    {
+        // Arrange & Act
+        var component = CompileToComponent($@"
+@using static Microsoft.AspNetCore.Components.DefaultRenderModes
+@rendermode Server
+");
+
+        // Assert
+        var attribute = Assert.Single(component.GetType().CustomAttributes);
+        Assert.Equal("PrivateComponentRenderModeAttribute", attribute.AttributeType.Name);
+
+        var nestedType = component.GetType().GetNestedType("PrivateComponentRenderModeAttribute", System.Reflection.BindingFlags.NonPublic);
+        Assert.NotNull(nestedType);
+
+        var modeProperty = nestedType.GetProperty("Mode");
+        Assert.NotNull(modeProperty);
+    }
+
+    [Fact]
+    public void RenderMode_Missing_Value()
+    {
+        // Arrange & Act
+        var compilationResult = CompileToCSharp($@"
+@rendermode
+");
+
+        // Assert
+        // Error RZ1014: The 'rendermode' directive expects a namespace name.
+        var diagnostic = Assert.Single(compilationResult.Diagnostics);
+        Assert.Equal("RZ1014", diagnostic.Id);
+    }
+
+    [Fact]
+    public void DuplicateRenderModes()
+    {
+        // Arrange & Act
+        var compilationResult = CompileToCSharp($@"
+@rendermode Microsoft.AspNetCore.Components.DefaultRenderModes.Server
+@rendermode Microsoft.AspNetCore.Components.DefaultRenderModes.Server
+");
+
+        // Assert
+        //Error RZ2001: The 'rendermode' directive may only occur once per document.
+        var diagnostic = Assert.Single(compilationResult.Diagnostics);
+        Assert.Equal("RZ2001", diagnostic.Id);
+    }
+
+    [Fact]
+    public void RenderMode_With_InvalidIdentifier()
+    {
+        var compilationResult = CompileToCSharp($@"
+@rendermode NoExist
+");
+
+        Assert.Empty(compilationResult.Diagnostics);
+
+        var assemblyResult = CompileToAssembly(compilationResult, throwOnFailure: false);
+        assemblyResult.Diagnostics.Verify(
+                    // x:\dir\subdir\Test\TestComponent.cshtml(1,58): error CS0103: The name 'NoExist' does not exist in the current context
+                    //             public override IComponentRenderMode Mode => NoExist;
+                    Diagnostic(ErrorCode.ERR_NameNotInContext, "NoExist").WithArguments("NoExist").WithLocation(1, 58));
+    }
+
+    [Fact]
+    public void LanguageVersion()
+    {
+        var compilationResult = CompileToCSharp($@"
+@rendermode Microsoft.AspNetCore.Components.DefaultRenderModes.Server
+", configuration: Configuration.WithVersion(RazorLanguageVersion.Version_7_0));
+
+        Assert.Empty(compilationResult.Diagnostics);
+
+        var assemblyResult = CompileToAssembly(compilationResult, throwOnFailure: false);
+        assemblyResult.Diagnostics.Verify(
+            // x:\dir\subdir\Test\TestComponent.cshtml(1,3): error CS0103: The name 'rendermode' does not exist in the current context
```

Comment:
If we move the RazorLanguageVersion into the `ComponentRenderModeDirectivePass`, can we make this a better error message easily?

Though actually this makes me realize this is actually a breaking change, as someone could have a variable called `rendermode` in their Razor files right now. I'm sure that was called out in the proposal, but I must have missed it.

Reply:
Huh, thats a really good point, and I don't think it was called out. I'll add it to the LDM thread.